src/index.js

```diff
@@ -228,7 +241,8 @@ export class Swipeable extends React.PureComponent {
 
   render() {
     const { className, style, nodeName = 'div', innerRef, children, trackMouse } = this.props
-    const handlers = getHandlers(this._set, { trackMouse })
+    const [handlers, attachTouch] = getHandlers(this._set, { trackMouse })
```

Comment:
💡 Thoughts on memoizing the class component?

Could add `handlers` and `attachTouch` to `this` and set in the constructor initially, and add a `componentDidUpdate` that only re-runs `getHandlers` if `trackMouse ` changes?

Reply:
It should work flawlessly.

Let me try implement this enhancement:

if it works, I'll push the changes here.

Reply:
## Cache handlers

We should use `getDerivedStateFromProps` instead of `componentDidUpdate` to preserve
the same order of operations between the hook version and the component version:

`componentDidUpdate`  runs in the commit phase while all the `useSwipeable` code runs in the render phase([see](https://twitter.com/dan_abramov/status/981712092611989509)).

Here's a snippet:

```javascript
 [...]
 static getDerivedStateFromProps({ trackMouse }, { trackMouse: prevTrackMouse, _set }) {
    if (prevTrackMouse !== trackMouse) {
      const [handlers, attachTouch] = getHandlers(_set, { trackMouse })

      return { trackMouse, handlers, attachTouch }
    }
    return null
  }
[...]
```

## SSR

We should add 1 or 2 SSR tests but I think it is outside the scope of this pull request.

Reply:
Pushed a version that uses getDerivedStateFromProps [in this branch](https://github.com/FaberVitale/react-swipeable/tree/cache-component-handlers).

I've noticed that this packages has React >= 16.0.0 as peerDependency but `getDerivedStateFromProps` got added in [v16.3](https://github.com/facebook/react/blob/master/CHANGELOG.md#1630-march-29-2018).

You'd probably have to bump to a major version if you want to either use [getDerivedStateFromProps](https://github.com/FaberVitale/react-swipeable/tree/cache-component-handlers) 
or move the logic of both hook and component logic from the render phase to the commit one.

Reply:
I agree with `getDerivedStatefromProps` and really like your implementation.

Release a major to include the class-based improvements is definitely the right way along with bumping the required minimum version of react.

With your guidance and example implementation, I think I'm leaning towards holding off, possibly indefinitely, on the including the class-based improvements. As hooks are the path forward I think the next major of this package should probably deprecate the component and only export a hook.

Let's just move forward with this amazing addition to the hook! Thank you @FaberVitale!